src/Servers/IIS/IIS/src/Core/IISHttpContext.cs

```diff
@@ -395,6 +395,13 @@ private async Task ProduceEndAwaited()
         await _bodyOutput.FlushAsync(default);
     }
 
+    // Response trailers, reset, and GOAWAY are only on HTTP/2+ and require IIS support
+    // that is only available on Win 11/Server 2022 or later.
+    private static readonly bool isCompatibleOsVersion = OperatingSystem.IsWindowsVersionAtLeast(10, 0, 20348, 0);
```

Comment:
I see 22000 as the official Win11 version. https://en.wikipedia.org/wiki/Windows_11_version_history
20303 might be the Server 2022 version? https://simple.wikipedia.org/wiki/Windows_Server_2022

Reply:
20303 was a preview. 20348 is the official release: https://docs.microsoft.com/en-us/windows-server/get-started/windows-server-release-info